### PR TITLE
Upgrade sdk to gemini-3g-2024-jan-08

### PR DESCRIPTION
**Pull request by ParthDesai**
# Description
This PR updates sdk to `gemini-3g-2023-jan-08` snapshot.

## Notable changes:
- Farm arguments are changed to be in parity with monorepo farmer's cli arguments.

### CODE REVIEW

**Comment by ParthDesai** [comment]
Mac CI jobs are failing with the error: `*** read jobs pipe: Resource temporarily unavailable`

**Comment by ParthDesai** [comment]
> Mac CI jobs are failing with the error: `*** read jobs pipe: Resource temporarily unavailable`. Investigating..
> 
> ```shell
> perl: warning: Falling back to the standard locale ("C").
> configure.ac:44: installing './config/compile'
> configure.ac:40: installing './config/missing'
> contrib/misc/Makefile.am: installing './config/depcomp'
> autoreconf: './config/config.sub' is updated
> autoreconf: './config/config.guess' is updated
> autoreconf: './config/install-sh' is updated
> autoreconf: Leaving directory '.'
> configure: WARNING: "-fvisibility=hidden" has been added to the hwloc CFLAGS
> make[1]: *** read jobs pipe: Resource temporarily unavailable.  Stop.
> make[1]: *** Waiting for unfinished jobs....
> make: *** [install-recursive] Error 1
> ```

Fixed now.

**Review by nazar-pc — COMMENTED** [review]
I think many of the things don't need `numa` feature on them

File: Cargo.toml [hunk 1]
```diff
@@ -150,7 +150,14 @@ members = [
...
 
 [features]
-default = []
+default = ["numa"]
+numa = [
+    "sdk-farmer/numa",
+    "sdk-dsn/numa",
+    "sdk-utils/numa",
+    "sdk-node/numa",
+    "sdk-substrate/numa"
```
[the remark this answers is not in the document]
Comment (ParthDesai) [reply Cargo.toml:159]: @nazar-pc Yes. Almost every package either uses something from subspace farmer (for example, utils uses `NodeClient` and dsn uses `PieceCache`) and/or uses the utils and dsn package.

Initial plan to have only `sdk-farmer` have `numa` feature, but there is enough inter-dependency that I need to add it everywhere.

[the remark this answers is not in the document]
Comment (ParthDesai) [reply Cargo.toml:159]: Hmm. Maybe I can mark `required-features=false` and try to simplify this.

[the remark this answers is not in the document]
Comment (nazar-pc) [reply Cargo.toml:159]: > Maybe I can mark default-features=false and try to simplify this

Yes, please do

**Review by nazar-pc — APPROVED** [review]

**Pull request merged**